sharding/nodesSetup_test.go

```diff
@@ -514,3 +673,464 @@ func TestNodesSetup_NodesAboveMaxShardLimit(t *testing.T) {
+
+	require.Equal(t, uint32(1), nrShards)
+}
+
+func TestNodesSetup_GetShardConsensusGroupSize(t *testing.T) {
+	t.Parallel()
+
+	noOfInitialNodes := 3
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 2,
+		shardMinNodes:      2,
+		metaConsensusSize:  1,
+		metaMinNodes:       1,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+	require.Nil(t, err)
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+
+	shardConsensusGroupSize := ns.GetShardConsensusGroupSize()
+
+	require.Equal(t, uint32(2), shardConsensusGroupSize)
+}
+
+func TestNodesSetup_GetMetaConsensusGroupSize(t *testing.T) {
+	t.Parallel()
+
+	noOfInitialNodes := 3
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 1,
+		shardMinNodes:      1,
+		metaConsensusSize:  2,
+		metaMinNodes:       2,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+	require.Nil(t, err)
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+
+	metaConsensusGroupSize := ns.GetMetaConsensusGroupSize()
+
+	require.Equal(t, uint32(2), metaConsensusGroupSize)
+}
+
+func TestNodesSetup_GetHysteresis(t *testing.T) {
+	t.Parallel()
+
+	ns := &NodesSetup{}
+	ns.Hysteresis = 0.5
+
+	hysteresis := ns.GetHysteresis()
+
+	require.Equal(t, float32(0.5), hysteresis)
+}
+
+func TestNodesSetup_GetAdaptivity(t *testing.T) {
+	t.Parallel()
+
+	ns := NodesSetup{}
+
+	ns.Adaptivity = true
+	adaptivity := ns.GetAdaptivity()
+
+	require.True(t, adaptivity)
+}
+
+func TestNodesSetup_GetStartTime(t *testing.T) {
+	t.Parallel()
+
+	ns := NodesSetup{}
+
+	ns.StartTime = 2
+	startTime := ns.GetStartTime()
+
+	require.Equal(t, int64(2), startTime)
+}
+
+func TestNodesSetup_GetRoundDuration(t *testing.T) {
+	t.Parallel()
+
+	ns := NodesSetup{}
+
+	ns.RoundDuration = 2
+	startTime := ns.GetRoundDuration()
+
+	require.Equal(t, uint64(2), startTime)
+}
+
+func TestNodesSetup_ExportNodesConfigShouldWork(t *testing.T) {
+	t.Parallel()
+
+	noOfInitialNodes := 3
+
+	ns, err := createTestNodesSetup(argsTestNodesSetup{
+		shardConsensusSize: 1,
+		shardMinNodes:      1,
+		metaConsensusSize:  1,
+		metaMinNodes:       1,
+		numInitialNodes:    3,
+		genesisMaxShards:   1,
+	})
+	ns.Hysteresis = 0.2
+	ns.Adaptivity = false
+	ns.StartTime = 10
+
+	require.Nil(t, err)
+
+	ns = createAndAssignNodes(ns, noOfInitialNodes)
+	configNodes := ns.ExportNodesConfig()
+
+	require.Equal(t, int64(10), configNodes.StartTime)
+
+	for i, nodeConfig := range configNodes.InitialNodes {
+		require.Equal(t, config.InitialNodeConfig{PubKey: pubKeys[i], Address: address[i], InitialRating: 0},
+			config.InitialNodeConfig{PubKey: nodeConfig.PubKey, Address: nodeConfig.Address, InitialRating: nodeConfig.InitialRating})
+	}
+
+}
+
+func TestNodesSetup_IsInterfaceNil(t *testing.T) {
+	t.Parallel()
+
+	ns := NodesSetup{}
+
+	ns.StartTime = 2
+	interfaceNil := ns.IsInterfaceNil()
+
+	require.False(t, interfaceNil)
```

Comment:
```suggestion
	ns, _ := NewNodesSetup(config.NodesConfig{}, nil, nil, nil, 0)
	require.True(t, ns.IsInterfaceNil())

	ns, _ = createTestNodesSetup(argsTestNodesSetup{
		shardConsensusSize: 1,
		shardMinNodes:      1,
		metaConsensusSize:  1,
		metaMinNodes:       1,
		numInitialNodes:    3,
		genesisMaxShards:   1,
	})
	require.False(t, ns.IsInterfaceNil())
```
this can be done in this way